Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/EOAWalletToEmbeddedWalletAdapter.cs

```diff
@@ -0,0 +1,252 @@
+                        {
+                            tokenIds[j] = BigInteger.Parse(tx.vals[j].id);
+                            tokenValues[j] = BigInteger.Parse(tx.vals[j].amount);
+
+                        }
+                        transaction = await erc1155.SafeBatchTransferFrom(Wallet.GetAddress(), toAddress, tokenIds, tokenValues, Encoding.UTF8.GetBytes(tx.data)).Create(client, new ContractCall(new Address(Wallet.GetAddress())));
+
+                        break;
+                    default:
+                        return new FailedTransactionReturn("Error adapting to EthTransaction type. Unable to determine transaction type for:" + transactions[i].ToString(), null);
+                }
+                
+                if (Wallet is Sequence.Wallet.EOAWallet wallet)
+                {
+                    if (waitForReceipt)
+                    {
+                        try
+                        {
+                            var responseReceipt = await Wallet.SendTransactionAndWaitForReceipt(client, transaction);
+
+                            if (responseReceipt == null)
+                            {
+                                throw new Exception("No Receipt");
+                            }
+                            else
+                            {
+                                var successfullReturn = new SuccessfulTransactionReturn(responseReceipt.transactionHash, null, null, null, JObject.FromObject(responseReceipt));
+                                successfulTransactionReturns.Add(successfullReturn);
+                                OnSendTransactionComplete?.Invoke(successfullReturn);
+                            }
+                        }
+                        catch (Exception receiptEx)
+                        {
+                            Debug.LogError("Error while waiting for the receipt: " + receiptEx.Message);
+                            string receiptExString = receiptEx.Message;
+                            var failedReturn = new FailedTransactionReturn(receiptEx.Message, null, null);
+                            failedTransactionReturns.Add(failedReturn);
+                            OnSendTransactionFailed?.Invoke(failedReturn);
+                        }
+                    }
+                    else
+                    {
+                        var responseReceipt = await Wallet.SendTransaction(client, transaction);
+                        successfulTransactionReturns.Add(new SuccessfulTransactionReturn(responseReceipt, null, null, null));
+                    }
+                }
+            }
+
+            if (failedTransactionReturns.Count == 0)
+            {
+                return successfulTransactionReturns.Count > 1
+                    ? new SuccessfulBatchTransactionReturn(successfulTransactionReturns.ToArray())
+                    : successfulTransactionReturns[0];
+            }
+
+            return failedTransactionReturns.Count > 1
```

Comment:
If one transaction fails but 10 pass, won't this only return the one failed transaction?